streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -526,6 +521,9 @@ public synchronized void close() {
         fOptions.close();
         filter.close();
         cache.close();
+        if (statistics != null) {
+            statistics.close();
+        }
 
```

Comment:
What happens if a user-specified statistics object is closed by user code?

Reply:
The user still needs to close the statistics he provided. This will just close the statistics handle.

I added a test for this. It doesn't check everything, but that is hard to do.

Reply:
Btw -- why/when would a user want to specify the statistics themselves vs just letting Streams handle it? 

And on a related note, is there any chance failing to close this (either by the user or by Streams) might be causing some of the recent reports of getting no results and just seeing `0.0` for a bunch of the rocksdb metrics? I just feel like I've started seeing these pop up again and can't tell if there's a bug in user code or somehow in ours

Reply:
Sorry, I meant the "real" docs on the actual kafka site, not the javadocs. Seems like it would be useful to show how/why to use your own `Statistics` like this. Anyways I just filed a ticket for this so someone else can pick it up https://issues.apache.org/jira/browse/KAFKA-14459